packages/discord.js/src/structures/Message.js

```diff
@@ -642,9 +642,9 @@ class Message extends Base {
    * @property {MessageMentionOptions} [allowedMentions] Which mentions should be parsed from the message content
    * @property {MessageFlags} [flags] Which flags to set for the message.
    * Only `MessageFlags.SuppressEmbeds` can be edited.
-   * @property {Attachment[]} [attachments] An array of attachments to keep,
+   * @property {Attachment[]|AttachmentBuilder[]} [attachments] An array of attachments to keep,
```

Comment:
I think the relevant portion for Discord here is the id of the attachment, which builders don't have:
```suggestion
   * @property {Attachment[]} [attachments] An array of attachments to keep,
```